runtime/caml/weak.h

```diff
@@ -19,6 +19,7 @@
 #define CAML_WEAK_H
 
 #include "mlvalues.h"
+#include "memory.h"
```

Comment:
Probably not for this PR specifically, but `weak.h` does use stuff from `memory.h` in `caml_ephe_clean`, so that there is a dependency between these two files.

Reply:
Ah, so it does! This has been wrong for a while then, and only worked by coincidence (the files that included "weak.h" happened to include "memory.h" beforehand). Thanks for the fix!